exchanges/kraken/kraken_test.go

```diff
@@ -67,6 +66,301 @@ func TestGetServerTime(t *testing.T) {
 	}
 }
 
+func TestFetchTradablePairs(t *testing.T) {
+	t.Parallel()
+	_, err := k.FetchTradablePairs(asset.Futures)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestUpdateTicker(t *testing.T) {
+	t.Parallel()
+	sp, err := currency.NewPairFromString("XBTUSD")
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = k.UpdateTicker(sp, asset.Spot)
+	if err != nil {
+		t.Error(err)
+	}
+
+	fp, err := currency.NewPairFromString("pi_xbtusd")
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = k.UpdateTicker(fp, asset.Futures)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestUpdateOrderbook(t *testing.T) {
+	t.Parallel()
+	sp, err := currency.NewPairFromString("BTCEUR")
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = k.UpdateOrderbook(sp, asset.Spot)
+	if err != nil {
+		t.Error(err)
+	}
+
+	fp, err := currency.NewPairFromString("pi_xbtusd")
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = k.UpdateOrderbook(fp, asset.Futures)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestUpdateAccountInfo(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip("skipping test: api keys not set")
+	}
+	_, err := k.UpdateAccountInfo(asset.Spot)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestWrapperGetOrderInfo(t *testing.T) {
+	if !areTestAPIKeysSet() {
+		t.Skip("skipping test: api keys not set")
+	}
+	t.Parallel()
+	_, err := k.GetOrderInfo("123", currency.Pair{}, asset.Futures)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestFuturesBatchOrder(t *testing.T) {
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip("skipping test: api keys not set or canManipulateRealOrders")
+	}
+	t.Parallel()
+	var data []PlaceBatchOrderData
+	var tempData PlaceBatchOrderData
+	tempData.PlaceOrderType = "cancel"
+	tempData.OrderID = "test123"
+	tempData.Symbol = "pi_xbtusd"
+	data = append(data, tempData)
+	_, err := k.FuturesBatchOrder(data)
+	if err != nil {
+		t.Error(err)
```

Comment:
So I accidentally ran this test with SPOT credentials and the function passed. Using verbose I can see the following:
```
HTTP status: 200 OK, Code: 200
Kraken raw response: {"result":"error","error":"authenticationError","serverTime":"2021-01-31T23:04:59.024Z"}
```
So this kind of situation has been implementated in other exchanges from shazbert. Under your kraken functions `SendAuthenticatedHTTPRequest` instead of json decoding into the passed in type, you instead have a
`interim := json.RawMessage{}`.
Then you look for errors inside the interim before decoding the json into the return type `return json.Unmarshal(intermediary, result)`. Huobi is a good example.

tl;dr ~~stonks only go up~~ please add some error checks to `SendAuthenticatedHTTPRequest` and `SendFuturesAuthRequest` to catch these errors